cmd/gateclient/client.go

```diff
@@ -368,6 +376,53 @@ func (m *GatewayClient) authenticateIAP() (string, error) {
 	return token, err
 }
 
+func (m *GatewayClient) authenticateGoogleServiceAccount() (err error) {
+	auth := m.Config.Auth
+	gsa := auth.GoogleServiceAccount
+
+	if !gsa.IsValid() {
+		return nil
+	}
+
+	if gsa.CachedToken != nil && gsa.CachedToken.Valid() {
+		return m.login(gsa.CachedToken.AccessToken)
+	}
+	gsa.CachedToken = nil
+
+	var source oauth2.TokenSource
+	if gsa.File == "" {
+		source, err = google.DefaultTokenSource(context.Background(), "profile", "email")
+	} else {
+		p, ferr := ioutil.ReadFile(gsa.File)
+		if ferr != nil {
+			return err
+		}
+		source, err = google.JWTAccessTokenSourceFromJSON(p, "https://accounts.google.com/o/oauth2/v2/auth")
```

Comment:
May also want to fetch this from the discovery document's `authorization_endpoint` value (see https://developers.google.com/identity/protocols/OpenIDConnect#discovery).